govcd/vm.go

```diff
@@ -302,6 +361,43 @@ func (vm *VM) RunCustomizationScript(computername, script string) (Task, error)
 	return vm.Customize(computername, script, false)
 }
 
+func (vm *VM) GetGuestCustomizationStatus() (string, error) {
+	guestCustomizationStatus := &types.GuestCustomizationStatusSection{}
+
+	if vm.VM.HREF == "" {
+		return "", fmt.Errorf("cannot load guest customization, VM HREF is empty")
+	}
+
+	_, err := vm.client.ExecuteRequest(vm.VM.HREF+"/guestcustomizationstatus", http.MethodGet,
+		types.MimeGuestCustomizationStatus, "error retrieving guest customization status: %s", nil, guestCustomizationStatus)
+
+	// The request was successful
+	return guestCustomizationStatus.GuestCustStatus, err
+}
+
+// BlockWhileGuestCustomizationStatus blocks until the customization status of VM exits unwantedStatus.
+// It sleeps 3 seconds between iterations and times out after timeOutAfterSeconds of seconds.
+func (vm *VM) BlockWhileGuestCustomizationStatus(unwantedStatus string, timeOutAfterSeconds int) error {
```

Comment:
Can we have validations for statuses and seconds?

Reply:
Not 100% sure is we really know all the possible statuses. As far as time I can, but probably that could be limited to simply checking if it is not zero, nothing more

Reply:
So let's add atleast time validation, to not create some unlimited one - constant loop

Reply:
Ok. What upper limit do we think makes sense? 30 mins?

Reply:
could be and also not less zero, I don't know how it would work with -30.

Reply:
Added to be 4<X<7200  (minimum is 4, because first check kicks in after 3 seconds)
I left  quite big maximum as I do remember having some big Windows "guest machine customization" happening > 1hour from my experience. Hope this is enough.

Reply:
Not a big fan of timeouts hidden deep inside the code... Is there a way to have a configurable timeout parameter which would be used as a common one for more than one function? This configuration parameter could be hidden from the user to avoid complexity, however, in case of user issue, we could ask to increase it for troubleshooting.

Reply:
The timeout is here. `timeOutAfterSeconds int` is a parameter, it just has the validator now as guys wanted. From usage perspective at the moment I only really needed it for tests so far, but one can change it.